File: backend-go/internal/model/user_dto.go
package model

import (
	"time"

	"agussyahrilmubarok.github.io/backend/internal/domain"
)

type UserResponse struct {
	ID        string    `json:"id"`
	Name      string    `json:"name"`
	Email     string    `json:"email"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

func ToUserResponse(user *domain.User) UserResponse {
	return UserResponse{
		ID:        user.ID.String(),
		Name:      user.Name,
		Email:     user.Email,
		CreatedAt: user.CreatedAt,
		UpdatedAt: user.UpdatedAt,
	}
}

type CreateUserRequest struct {
	Name     string `json:"name" binding:"required,min=2,max=100"`
	Email    string `json:"email" binding:"required,email,max=100"`
	Password string `json:"password" binding:"required,min=8,max=255"`
}

type UpdateUserRequest struct {
	Name     string `json:"name" binding:"omitempty,min=2,max=100"`
	Email    string `json:"email" binding:"omitempty,email,max=100"`
	Password string `json:"password" binding:"omitempty,min=8,max=255"`
}

File: backend-dotnet/Backend/Application/DTOs/UserDto.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Backend.Application.DTOs;

public sealed record CreateUserRequest(
    [property: JsonPropertyName("name")]
    [Required]
    [MinLength(2)]
    [MaxLength(100)]
    string Name,
    [property: JsonPropertyName("email")]
    [Required]
    [EmailAddress]
    [MaxLength(150)]
    string Email,
    [property: JsonPropertyName("password")]
    [Required]
    [MinLength(8)]
    [MaxLength(72)]
    string Password
);

public sealed record UpdateUserRequest(
    [property: JsonPropertyName("name")]
    [MinLength(2)]
    [MaxLength(100)]
    string? Name,
    [property: JsonPropertyName("email")]
    [EmailAddress]
    [MaxLength(150)]
    string? Email,
    [property: JsonPropertyName("password")]
    [MinLength(8)]
    [MaxLength(72)]
    string? Password
);

public sealed record UserResponse(
    [property: JsonPropertyName("id")] string Id,
    [property: JsonPropertyName("name")] string Name,
    [property: JsonPropertyName("email")] string Email,
    [property: JsonPropertyName("created_at")]
    DateTimeOffset CreatedAt,
    [property: JsonPropertyName("updated_at")]
    DateTimeOffset UpdatedAt
);